### ISSUE TITLE
PowerMod for univariate polynomials yields incorrect results

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#146]
The `PowerMod` method for univariate polynomials, `POWMOD_UPOLY`, apparently sometimes returns incorrect results. Consider this example:
```
p := 227;; x := X(GF(p), "x");; f := x^(7^2) - x;;
```
Then compare these results:
```
gap> PowerMod(x, p, f);
x
gap> (x^p) mod f;
x^35
gap> QuotientRemainder(x ^ p, f);
[ x^178+x^130+x^82+x^34, x^35 ]
```
So `x^35` is the right answer, but `PowerMod` claims it is `x`.

Ouch.

**Comment by stevelinton** [comment]
The bug is in POWMOD_UPOLY in ratfunul.gi. I think the correct fix is to delete lines 727 to 731 and 735 which would be correct for addition or subtraction but are definitely wrong for powermod. This is not my code, so I'm not sure if this is right, and there might be other bugs of a similar flavour.

**Comment by alex-konovalov** [comment]
Fixed in stable-4.7 branch in PR #152. Should be propagated to the master branch in the next periodic merge of stable-4.7 into master.
